### PR TITLE
refactor: adds business category, refactors onboarding structs

### PR DESCRIPTION
**Pull request by ontje-dev**
see: https://docs.mollie.com/overview/common-data-types#business-category

# Description

- Updating the onboarding Feature
  - Created new business categories replacing the CategoryCode which were deprecated in 2022.
  - Moved Onboarding structs to own definition for better use

## Motivation and context

Thanks for this work - i used it for onboarding in one of my projects.
The CategoryCode is no longer supported by mollie.

## How has this been tested?

Existing tests were updated to use the new business category.

- [x] Unit tests updated

Tests running locally on:

- OS: Windows 10
- Go version [go1.18.2 windows/amd64]

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.

Please, please, please, don't send your pull request until all of the boxes are ticked. Once your pull request is created, it will trigger a build on our continuous integration server to make sure your [tests and code style pass](https://help.github.com/articles/about-required-status-checks/).

- [x] I have read the **[CONTRIBUTING](CONTRIBUTING.md)** document.
- [x] My pull request addresses exactly one patch/feature.
- [x] I have created a branch for this patch/feature.
- [x] Each individual commit in the pull request is meaningful.
- [x] If my change requires a change to the documentation, I have updated it accordingly.

### CODE REVIEW

**Comment by VictorAvelar** [comment]
Hey @ontje-dev , thanks for the contribution. I somehow missed the deprecation notice.

There are a couple of changes that I would like to suggest (I will open the conversations in the code), as well as offering an explanation for the reason I believe those changes are required.

It looks great overall and with minimal changes I am sure we can ship it.

File: mollie/common_types.go [hunk 1]
```diff
@@ -115,32 +115,147 @@ type PaginationLinks struct {
 	Documentation *URL `json:"documentation,omitempty"`
 }
 
-// CategoryCode specifies an industry or category.
-type CategoryCode uint
```
Comment (VictorAvelar) [inline mollie/common_types.go:119]: Existing implementations of a type would be broken if we decided to remove it. I've always agreed that the API layer should be responsible for data validation and accuracy.

As a result, I recommend keeping the type but changing the type comment to state that it shouldn't be used any longer.

That way we create a new type without completely breaking existing implementations. WDYT?

File: mollie/common_types.go [hunk 2]
```diff
@@ -115,32 +115,147 @@ type PaginationLinks struct {
...
+// BusinessCategory specifies an industry or category.
+type BusinessCategory string
 
-// Available category codes.
+// Available business categoreis.
```
Comment (VictorAvelar) [inline mollie/common_types.go:121]: A typo in the comment.

```suggestion
// Available business categories.
```

File: mollie/common_types.go [hunk 3]
```diff
@@ -115,32 +115,147 @@ type PaginationLinks struct {
...
 const (
-	BookMagazinesAndNewspapers          CategoryCode = 5192
```
Comment (VictorAvelar) [inline mollie/common_types.go:123]: Similar to the 'CategoryCode' type, I want to maintain these while changing the block comment to say that they are "historic" and shouldn't be used any longer.

File: mollie/onboarding.go [hunk 4]
```diff
@@ -62,22 +62,24 @@ func (os *OnboardingService) GetOnboardingStatus(ctx context.Context) (res *Resp
...
 //
 // Information that the merchant has entered in their dashboard will not be overwritten.
+type OnboardingDataOrganization struct {
+	Name               string   `json:"name,omitempty"`
+	Address            *Address `json:"address,omitempty"`
+	RegistrationNumber string   `json:"registrationNumber,omitempty"`
+	VatNumber          string   `json:"vatNumber,omitempty"`
+	VatRegulation      string   `json:"vatRegulation,omitempty"`
+}
+type OnboardingDataProfile struct {
```
Comment (VictorAvelar) [inline mollie/onboarding.go:72]: A comment for each struct must be provided describing the purpose and content of it.

What I usually do is try to use the "docs" phrasing and just make tiny changes to fit the implementation.

**Review by VictorAvelar — CHANGES_REQUESTED** [review]
If you have any concerns or doubts about the code review, do let me know.

I want to thank you again for your incredible contribution.

**Comment by ontje-dev** [comment]
Hi @VictorAvelar , thanks for the review.
I aggree on every point and tried to fix them :)

**Comment by VictorAvelar** [comment]
Thanks for the updates @ontje-dev, I will merge this one to a staging branch to fix a couple of linter errors and then will be merged and released to the `master` branch.

I really appreciate you taking the time to put in such an amazing and sizeable contribution. 🤗

**Review by VictorAvelar — APPROVED** [review]

**Pull request merged**